### ISSUE TITLE
Max Average Approach diverges in cifar10

### ISSUE THREAD

**Issue by SeokilHam** [TAMU-VITA/triple-wins#4]
Hello, I want to use your code for my experiment, but I cannot train ResNet and MobileNet with your max average approach.

I discovered that a model(ResNet or MobileNet) diverges when I train a model with Max Average Approach using your code.

I think that there are some experiment details what you did not mention in your paper such as batch size or warm-up in ResNet experiments.

I want to know specific experiment details. Can you tell me more specific experiments details?

Also, in MobileNet, there are 7 branches in your code although 3 branches in your paper.

I also want to use MobileNet v2 with 3 branches version code. 
Can you upload your 3 branches version code if you allow to open to the public?

Thank you

**Comment by tingkuei** [comment]
Hi SeoKilHam,

If you want to train with the max-average loss. You could set "--defend_algo ifgm_max_v2". For the resnet experiments, I didn't tune for the batch size and learning rate. Could you indicate what exactly the divergence looks like? 

For the mobileNet, actually only 3 branches are used during training and testing although I specify 7 branches in the model definition. That's just because I want to re-use the script for training.
You could try "python3 train_base_mob.py train cifar10_mobilenetv2_1-d cifar10 --attack_algo None --defend_algo ifgm_max_v2 --save-folder /path/to/folder"

**Comment by SeokilHam** [comment]
Thank you for fast response.

Divergence means that my model's validation loss increases while the model's train loss decreases.
Even the model's train accuracy is near 100%, but the model's validation loss is 25~30%.
Overfitting is severe.

In mobileNet, I cannot find the code which limits other branches except for three branches.
Can you tell me which sentence code limits other branches to be trained?

Thank you!
